lc-hax/Scripts/Components/RigidbodyMovement.cs

```diff
@@ -1,103 +1,143 @@
+            if (Helper.LocalPlayer is not PlayerControllerB localPlayer) return;
+            this.gameObject.layer = localPlayer.gameObject.layer;
         }
-    }
-
-    void UpdateSprintMultiplier(Keyboard keyboard) {
-        this.SprintMultiplier =
-            keyboard.shiftKey.IsPressed()
-                ? Mathf.Min(this.SprintMultiplier + (5.0f * Time.deltaTime), 5.0f)
-                : 1.0f;
-    }
 
-    void Move(Vector3 direction) {
-        if (this.rigidbody.Unfake() is not Rigidbody rigidbody) return;
+        // Adjust collision box in Awake
+        const float adjustedWidth = 0.0f; // Adjust as needed
+        const float adjustedHeight = 0.0f; // Adjust as needed
+        const float adjustedDepth = -0.5f; // Adjust as needed
+        void Awake() {
+            this.characterController = this.gameObject.AddComponent<CharacterController>();
 
-        Vector3 forward = this.transform.forward;
-        Vector3 right = this.transform.right;
+            this.transform.localScale = new Vector3(adjustedWidth, this.transform.localScale.y, adjustedDepth);
+            this.characterController.height = 0.0f; // Adjust as needed
+            this.characterController.center = new Vector3(0f, 0.3f, 0.5f); // Adjust as needed
 
-        forward.y = 0.0f;
-        right.y = 0.0f;
-        forward = forward.normalized;
-        right = right.normalized;
+            this.keyboard = Keyboard.current;
+        }
 
-        Vector3 translatedDirection = (right * direction.x) + (forward * direction.z);
-        rigidbody.velocity += translatedDirection * Time.deltaTime * RigidbodyMovement.baseSpeed * this.SprintMultiplier;
-    }
+        // Update is called once per frame
+        void Update() {
+            // Read movement input from keyboard
+            Vector2 moveInput = new Vector2(this.keyboard.dKey.ReadValue() - this.keyboard.aKey.ReadValue(),
+                                            this.keyboard.wKey.ReadValue() - this.keyboard.sKey.ReadValue()).normalized;
+
+            // Apply speed modifier based on left control key
+            float speedModifier = 1.0f;
+            if (this.keyboard.leftCtrlKey.isPressed) {
+                speedModifier = controlSlowdoneMultiplier;
+            }
+
+            // Calculate movement direction relative to character's forward direction
+            Vector3 forward = Vector3.ProjectOnPlane(this.transform.forward, Vector3.up).normalized;
+            Vector3 right = Vector3.ProjectOnPlane(this.transform.right, Vector3.up).normalized;
+            Vector3 moveDirection = (forward * moveInput.y) + (right * moveInput.x);
+            moveDirection.y = 0f; // Remove vertical component from the movement direction
+
+            // Apply speed and sprint modifiers
+            moveDirection *= this.isSprinting ? baseSpeed * sprintSpeedMultiplier * speedModifier : baseSpeed * speedModifier;
+
+            // Toggle gravity with Q key
+            if (this.keyboard.qKey.wasPressedThisFrame) {
```

Comment:
Could you move this key into `InputListener` so that others can have a more centralised file to swap out keys? The rest of the keys should be fine. I just think non-standard keys should be there because we don’t know if they use `Q` for their microphone.